File: app/src/main/java/pavel/project/kotlin/kotlinnexample/dependencyinjection/components/AppComponent.java
package pavel.project.kotlin.kotlinnexample.dependencyinjection.components;

import android.content.Context;

import dagger.BindsInstance;
import dagger.Component;

/*@Singleton
@Component(modules = {AppModule.class})*/
public interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder context(Context context);

        AppComponent build();
    }

   // void inject(App app);
}

File: app/src/main/java/pavel/project/kotlin/kotlinnexample/dependencyinjection/components/AppComponent.kt
package pavel.project.kotlin.kotlinnexample.dependencyinjection.components

import dagger.Component
import pavel.project.kotlin.kotlinnexample.dependencyinjection.modules.NetworkModule
import pavel.project.kotlin.kotlinnexample.presentation.mvp.presenter.PresenterMain
import javax.inject.Singleton



@Singleton
@Component(modules = [NetworkModule::class])
interface AppComponent {
    fun inject(presenterMain: PresenterMain)
}
